### DISCUSSION TITLE
How to use utcnow() with server_onupdate

### DISCUSSION THREAD

**Discussion by alexted** [sqlalchemy/sqlalchemy#10438] [Usage Questions] [answered]
Hello! 
Could you please tell me if there is an option to use server_onupdate with this [function](https://docs.sqlalchemy.org/en/20/core/compiler.html#utc-timestamp-function)?

my table:
```python
user = Table(
    "user",
    metadata,
    Column("id", Integer, primary_key=True),
    Column("first_name", String),
    Column("last_name", String),
    Column("created_at", DateTime, nullable=False, server_default=utcnow()),
    Column("updated_at", DateTime, server_onupdate=utcnow()),  # ???
)
```

The documentation says that this attribute expects to accept a value of _**FetchedValue**_ type, but after reading I didn't understand how to apply it to this [function](https://docs.sqlalchemy.org/en/20/core/compiler.html#utc-timestamp-function).
Could you help me pls?

**Comment by zzzeek** [comment, marked as answer]
database DDL does not usually have an "on update" field for table columns, instead you need to use triggers that you would configure separately.    using triggers tends to be a more complex affair, see examples at https://www.postgresql.org/docs/current/sql-createtrigger.html .

I would consider not trying to have a server-defined function here and instead use SQLAlchemy's `onupdate`; you can still use the database-side `utcnow()` function, SQLAlchemy would add it to UPDATE queries:

```py
user = Table(
    "user",
    metadata,
    Column("id", Integer, primary_key=True),
    Column("first_name", String),
    Column("last_name", String),
    Column("created_at", DateTime, nullable=False, server_default=utcnow()),
    Column("updated_at", DateTime, onupdate=func.utcnow()), 
)
```

see https://docs.sqlalchemy.org/en/20/core/defaults.html#client-invoked-sql-expressions for this feature

**Reply by alexted** [reply]
thank you

**Reply by ScholliYT** [reply]
Note that `onupdate=func.utcnow()` will not update the `updated_at` if you make changes to your data through other ways than SQLAlchemy if I understand correctly.
